### ISSUE TITLE
Consul web-ui services search-bar, when start typing, loses focus on every character input

### ISSUE THREAD

**Issue by dmclf** [hashicorp/consul#20109]
#### Overview of the Issue

Consul web-ui services search-bar, when start typing, loses focus on every character input.
so searching for... for example: `hashicorp`
(9 characters) would require clicking with mouse back to the search-bar to focus 9 times)

#### Reproduction Steps

Steps to reproduce this issue:

1. Go to Consul-UI
2. Go to services
3. Type something in the search-bar.
![image](https://github.com/hashicorp/consul/assets/19852218/bb466b34-8090-48dd-9096-ae5a486a4410)
4. try to type hashicorp in one stretch.
![image](https://github.com/hashicorp/consul/assets/19852218/797bb4bf-7c61-4607-a6a0-6945e9cf1ca0)
5. above in my case requires regaining focus after every character input.

### Consul info for Server

```
build:
	prerelease = 
	revision = 9bcafa24
	version = 1.17.1
	version_metadata = 
runtime:
	arch = amd64
	cpu_count = 1
	goroutines = 1227
	max_procs = 1
	os = linux
	version = go1.20.12
runtime:
	arch = arm64
	cpu_count = 2
	goroutines = 498
	max_procs = 2
	os = linux
	version = go1.20.12

```

### Operating system and Environment details

OS: Ubuntu 22.04.3
Architecture: x86_64 + aarch64

dpkg info:
```
ii  consul         1.17.1-1     amd64        Consul is a distributed, highly available, and data center aware solution to connect and configure applications across dynamic, distributed infrastructure.
ii  consul         1.17.1-1     arm64        Consul is a distributed, highly available, and data ce>
```

**Comment by absolutelightning** [comment]
Thanks for reporting this @dmclf.
I looked into this. Here's what I have found. In the PR merged by @chris-hut - https://github.com/hashicorp/consul/pull/19226/

A dependency was introduced named - `ember-a11y-refocus` - because of `hashicorp/design-system-components` - https://github.com/hashicorp/consul/pull/19226/files#diff-3a968206d6de2fecfc5dacd7d94bab7744c9f5d5c999a816164d95cbc135c316R1871-R1888

In this dependency there is a js code which actually changes the focus to a div with id - `ember-a11y-refocus-nav-message`

If the id is not used directly in consul - we can add `display:none` on this div to allow [autofocus on search box](https://github.com/hashicorp/consul/blob/5fb9df1640cfa1aaf96228da1f44b31ecd6b825a/ui/packages/consul-ui/app/components/freetext-filter/index.hbs#L20).

I have opened a PR for your review if this change sounds ok.

![Screenshot 2024-01-08 at 4 00 13 PM](https://github.com/hashicorp/consul/assets/134911583/ea817d4d-cc39-4bce-b178-3da209df64f5)

cc @WenInCode

**Comment by WenInCode** [comment]
Thank you for the report, and thank you @absolutelightning with the proposed fix. I will take some time to go over this afternoon. 

Much appreciated!

**Comment by WenInCode** [comment]
The fix for this has been merged and will be released as part of the upcoming 1.17 patch release.
